### ISSUE TITLE
unresolved dependency: soot-all-in-one

### ISSUE THREAD

**Issue by jonas-db** [Ucombinator/shimple-analyzer#7]
I get the following error when i executed _sbt assembly_

> [warn]   https://repo1.maven.org/maven2/org/ucombinator/soot/soot-all-in-one/nig
> htly.20150205/soot-all-in-one-nightly.20150205.pom
> [info] Resolving jline#jline;2.12.1 ...
> [warn]  ::::::::::::::::::::::::::::::::::::::::::::::
> [warn]  ::          UNRESOLVED DEPENDENCIES         ::
> [warn]  ::::::::::::::::::::::::::::::::::::::::::::::
> [warn]  :: org.ucombinator.soot#soot-all-in-one;nightly.20150205: not found
> [warn]  ::::::::::::::::::::::::::::::::::::::::::::::
> [warn]
> [warn]  Note: Unresolved dependencies path:
> [warn]          org.ucombinator.soot:soot-all-in-one:nightly.20150205 (...\jaam\src\serializer\build.sbt#L4-9)
> [warn]            +- org.ucombinator:jaam-serializer_2.11:0.1-SNAPSHOT
> sbt.ResolveException: unresolved dependency: org.ucombinator.soot#soot-all-in-on
> e;nightly.20150205: not found
>         at sbt.IvyActions$.sbt$IvyActions$$resolve(IvyActions.scala:291)
>         at sbt.IvyActions$$anonfun$updateEither$1.apply(IvyActions.scala:188)
>         at sbt.IvyActions$$anonfun$updateEither$1.apply(IvyActions.scala:165)
>         at sbt.IvySbt$Module$$anonfun$withModule$1.apply(Ivy.scala:155)
>         at sbt.IvySbt$Module$$anonfun$withModule$1.apply(Ivy.scala:155)
>         at sbt.IvySbt$$anonfun$withIvy$1.apply(Ivy.scala:132)
>         at sbt.IvySbt.sbt$IvySbt$$action$1(Ivy.scala:57)
>         at sbt.IvySbt$$anon$4.call(Ivy.scala:65)
>         at xsbt.boot.Locks$GlobalLock.withChannel$1(Locks.scala:93)
>         at xsbt.boot.Locks$GlobalLock.xsbt$boot$Locks$GlobalLock$$withChannelRet
> ries$1(Locks.scala:78)
>         at xsbt.boot.Locks$GlobalLock$$anonfun$withFileLock$1.apply(Locks.scala:
> 97)
>         at xsbt.boot.Using$.withResource(Using.scala:10)
>         at xsbt.boot.Using$.apply(Using.scala:9)
>         at xsbt.boot.Locks$GlobalLock.ignoringDeadlockAvoided(Locks.scala:58)
>         at xsbt.boot.Locks$GlobalLock.withLock(Locks.scala:48)
>         at xsbt.boot.Locks$.apply0(Locks.scala:31)
>         at xsbt.boot.Locks$.apply(Locks.scala:28)
>         at sbt.IvySbt.withDefaultLogger(Ivy.scala:65)
>         at sbt.IvySbt.withIvy(Ivy.scala:127)
>         at sbt.IvySbt.withIvy(Ivy.scala:124)
>         at sbt.IvySbt$Module.withModule(Ivy.scala:155)
>         at sbt.IvyActions$.updateEither(IvyActions.scala:165)
>         at sbt.Classpaths$$anonfun$sbt$Classpaths$$work$1$1.apply(Defaults.scala
> :1369)
>         at sbt.Classpaths$$anonfun$sbt$Classpaths$$work$1$1.apply(Defaults.scala
> :1365)
>         at sbt.Classpaths$$anonfun$doWork$1$1$$anonfun$87.apply(Defaults.scala:1
> 399)
>         at sbt.Classpaths$$anonfun$doWork$1$1$$anonfun$87.apply(Defaults.scala:1
> 397)
>         at sbt.Tracked$$anonfun$lastOutput$1.apply(Tracked.scala:37)
>         at sbt.Classpaths$$anonfun$doWork$1$1.apply(Defaults.scala:1402)
>         at sbt.Classpaths$$anonfun$doWork$1$1.apply(Defaults.scala:1396)
>         at sbt.Tracked$$anonfun$inputChanged$1.apply(Tracked.scala:60)
>         at sbt.Classpaths$.cachedUpdate(Defaults.scala:1419)
>         at sbt.Classpaths$$anonfun$updateTask$1.apply(Defaults.scala:1348)
>         at sbt.Classpaths$$anonfun$updateTask$1.apply(Defaults.scala:1310)
>         at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
>         at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
>         at sbt.std.Transform$$anon$4.work(System.scala:63)
>         at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:22
> 6)
>         at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:22
> 6)
>         at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
>         at sbt.Execute.work(Execute.scala:235)
>         at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:226)
>         at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:226)
>         at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestric
> tions.scala:159)
>         at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
>         at java.util.concurrent.FutureTask.run(Unknown Source)
>         at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
>         at java.util.concurrent.FutureTask.run(Unknown Source)
>         at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
>         at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
>         at java.lang.Thread.run(Unknown Source)
> [error](serializer/*:update) sbt.ResolveException: unresolved dependency: org.u
> combinator.soot#soot-all-in-one;nightly.20150205: not found
> [error] Total time: 4 s, completed 30-sep-2016 15:49:32

**Comment by CAHollenbeck** [comment]
Okay, thanks. We're looking into it.

**Comment by adamsmd** [comment]
@jonas-db: It looks like it was a transient problem.  Can you try again with a fresh checkout?

**Comment by jonas-db** [comment]
I just did a completely new clone of the repository, but the problem remains. I guess the nightly build might not be available anymore? Can i use another version?

**Comment by adamsmd** [comment]
@jonas-db: The issue should be fixed now.  Reopen the issue if it doesn't work for you.

This appears to have been a latent issue for a while.  The `resolvers` setting from the top-level `build.sbt` was not being propagated into the sub-project `build.sbt` files.  It was fixed by moving `resolvers += ...` into `commonSettings`.  We didn't notice it because our developers had the needed dependencies in their `ivy2` cache.  I was able to reproduce the error by clearing that cache.  (This also explains the intermittent issues @CAHollenbeck had reproducing the error.)
